internal/controller/controlplaneauth/controller.go

```diff
@@ -157,9 +156,9 @@ func (c *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 		return managed.ExternalCreation{}, errors.New(errNotControlPlaneAuth)
 	}
 
-	a, err := c.accounts.Get(ctx, c.profile.ID)
+	userID, err := upclient.ExtractUserIDFromToken(c.profile.Session)
 	if err != nil {
-		return managed.ExternalCreation{}, errors.Wrap(err, "unable to retrieve account details: the token provided in the providerConfig is not a user token")
+		return managed.ExternalCreation{}, errors.Wrap(err, "unable to retrieve userId from token")
 	}
 
 	t, err := c.tokens.Create(ctx, &tokens.TokenCreateParameters{
```

Comment:
@muvaf we can now create an token for kubeconfig from UserToken or PersonalAccessToken - and map the TokenCreateParameters to the UserID